x-pack/plugins/elastic_assistant/server/lib/attack_discovery/graphs/default_attack_discovery_graph/index.ts

```diff
@@ -0,0 +1,122 @@
+import type { CompiledStateGraph } from '@langchain/langgraph';
+import { END, START, StateGraph } from '@langchain/langgraph';
+
+import { NodeType } from './constants';
+import { getGeneratOrEndEdge } from './edges/generate_or_end';
+import { getGeneratOrRefineOrEndEdge } from './edges/generate_or_refine_or_end';
+import { getRefineOrEndEdge } from './edges/refine_or_end';
+import { getRetrieveAnonymizedAlertsOrGenerateEdge } from './edges/retrieve_anonymized_alerts_or_generate';
+import { getDefaultGraphState } from './state';
+import { getGenerateNode } from './nodes/generate';
+import { getRefineNode } from './nodes/refine';
+import { getRetrieveAnonymizedAlertsNode } from './nodes/retriever';
+import type { GraphState } from './types';
+
+export interface GetDefaultAttackDiscoveryGraphParams {
+  alertsIndexPattern?: string;
+  anonymizationFields: AnonymizationFieldResponse[];
+  esClient: ElasticsearchClient;
+  llm: ActionsClientLlm | AssistantToolLlm;
+  logger?: Logger;
+  onNewReplacements?: (replacements: Replacements) => void;
+  replacements?: Replacements;
+  size: number;
+}
+
+export type DefaultAttackDiscoveryGraph = ReturnType<typeof getDefaultAttackDiscoveryGraph>;
+
+/**
+ * This function returns a compiled state graph that represents the default
+ * Attack discovery graph.
+ *
+ * Refer to the following diagram for this graph:
+ * x-pack/plugins/elastic_assistant/docs/img/default_attack_discovery_graph.png
+ */
+export const getDefaultAttackDiscoveryGraph = ({
+  alertsIndexPattern,
+  anonymizationFields,
+  esClient,
+  llm,
+  logger,
+  onNewReplacements,
+  replacements,
+  size,
+}: GetDefaultAttackDiscoveryGraphParams): CompiledStateGraph<
+  GraphState,
+  Partial<GraphState>,
+  'generate' | 'refine' | 'retrieve_anonymized_alerts' | '__start__'
+> => {
+  try {
+    const graphState = getDefaultGraphState();
+
+    // get nodes:
+    const retrieveAnonymizedAlertsNode = getRetrieveAnonymizedAlertsNode({
+      alertsIndexPattern,
+      anonymizationFields,
+      esClient,
+      logger,
+      onNewReplacements,
+      replacements,
+      size,
+    });
+
+    const generateNode = getGenerateNode({
+      llm,
+      logger,
+    });
+
+    const refineNode = getRefineNode({
+      llm,
+      logger,
+    });
+
+    // get edges:
+    const generateOrEndEdge = getGeneratOrEndEdge(logger);
+
+    const generatOrRefineOrEndEdge = getGeneratOrRefineOrEndEdge(logger);
+
+    const refineOrEndEdge = getRefineOrEndEdge(logger);
+
+    const retrieveAnonymizedAlertsOrGenerateEdge =
+      getRetrieveAnonymizedAlertsOrGenerateEdge(logger);
+
+    // create the graph:
+    const graph = new StateGraph<GraphState>({ channels: graphState })
+      .addNode(NodeType.RETRIEVE_ANONYMIZED_ALERTS_NODE, retrieveAnonymizedAlertsNode)
+      .addNode(NodeType.GENERATE_NODE, generateNode)
+      .addNode(NodeType.REFINE_NODE, refineNode)
+      .addConditionalEdges(START, retrieveAnonymizedAlertsOrGenerateEdge, {
```

Comment:
Consider following the `stepRouter` pattern in `getDefaultAssistantGraph` for the `addConditionalEdges` for improved readability

Reply:
As discussed in an offline chat with @stephmilovic, I aimed for consistency with the assistant graph and adopted the useful diagram tooling etc, but it didn't feel like the `stepRouter` abstraction improved readability in this case.